vignettes/flowmaps-static.qmd

```diff
@@ -0,0 +1,774 @@
+  id       census_districts                                              municipalities_mitma municipalities         district_names_in_v2 district_ids_in_v2                      geom
+  <chr>    <chr>                                                         <chr>                <chr>                  <chr>                <chr>                     <MULTIPOLYGON [m]>
+1 2408910  2408910                                                       24089                24089                  León distrito 10     2408910            (((290940.1 4719080, 290…
+2 22117_AM 2210201; 2210301; 2211501; 2211701; 2216401; 2218701; 2221401 22117_AM             22102; 22103; 22115; … Graus agregacion de… 22117_AM           (((774184.4 4662153, 774…
+3 2305009  2305009                                                       23050                23050                  Jaén distrito 09     2305009            (((429745 4179977, 42971…
+4 07058_AM 0701901; 0702501; 0703401; 0705801; 0705802                   07058_AM             07019; 07025; 07034; … Selva agregacion de… 07058_AM           (((1000859 4415059, 1000…
+5 2305006  2305006                                                       23050                23050                  Jaén distrito 06     2305006            (((429795.1 4180957, 429…
+6 2305005  2305005                                                       23050                23050                  Jaén distrito 05     2305005            (((430022.7 4181101, 429…
+```
+
+## Aggregate data - count total flows
+
+```{r}
+od_20210407_total <- od_20210407 |>
+  group_by(id_origin, id_destination) |>
+  summarise(n_trips = sum(n_trips, na.rm = TRUE)) |> 
+  collect()
+
+head(od_20210407_total)
+```
+
+```
+# A tibble: 6 × 3
+# Groups:   id_origin [2]
+  id_origin id_destination  n_trips
+  <fct>     <fct>             <dbl>
+1 0704003   07031           3958.  
+2 0704003   07036          17278.  
+3 0704003   07044            572.  
+4 0704003   0801903            6.19
+5 0704004   07027           3036.  
+6 0704004   0704003        56604.
+```
+
+## Reshape flows for visualization
+
+We are going to use the `{flowmapper}` [@flowmapper-r] package to plot the flows. This package expects the data to be in the following format:
+
+```{r}
+'data.frame':	7 obs. of  8 variables:
+ $ id_a   : chr  "X1" "X2" "X3" "X3" ...
+ $ id_b   : chr  "X5" "X4" "X1" "X5" ...
+ $ xa     : num  2 14 10 10 2 14 14
+ $ ya     : num  6 10 9 9 6 10 10
+ $ xb     : num  10 4 2 10 4 10 10
+ $ yb     : num  4 10 6 4 10 4 9
+ $ flow_ab: num  1 2 3 3 1 1 4
+ $ flow_ba: num  2 3 2 5 2 1 5
+```
+
+### Reshape the flows
+
+The data we have right now in `od_20210407_total` is in a classic long format. That is, we have rows for both the number of trips from A to B and from B to A. `{flowmapper}` requires the data to be in different format, where there is only one row for each pair of `id_a` and `id_b` and two columns, one for the flow from `id_a` to `id_b` and one for the flow from `id_b` to `id_a`.
```

Comment:
Is this something that can be done with `{od}`? I found the function to convert from long to oneway, but it is not the same...

Reply:
This is not relevant anymore for the particular vignette, as I am now using the dev version of https://github.com/JohMast/flowmapper/issues/3 that supports the long OD format. But may still be something that may be implemented in `{od}`.

Reply:
Result!